src/transformers/quantizers/quantizer_compressed_tensors.py

```diff
@@ -116,17 +120,19 @@ def _process_model_before_weight_loading(self, model, **kwargs):
 
         ct_quantization_config = self.compressor.quantization_config
 
-        if self.run_compressed:
-            if not self.is_quantization_compressed:
-                raise ValueError("`run_compressed` is only supported for quantized_compressed models")
-            apply_quantization_config(model, ct_quantization_config, run_compressed=True)
-        elif self.is_quantized and not self.is_quantization_compressed:
-            apply_quantization_config(model, ct_quantization_config)
+        update_structure_before_weight_load = self.run_compressed or (
+            self.quantization_config.is_quantized and not self.quantization_config.is_quantization_compressed
+        )
+
+        if update_structure_before_weight_load:
+            apply_quantization_config(model, ct_quantization_config, run_compressed=self.run_compressed)
```

Comment:
would self.run_compressed ever not be true here?  Why not just use that explicitly?

Reply:
For this specific line,  run_compressed can be True for a quantized compressed model